File: python/src/subscriber/subscriber_msg.py
#!/usr/bin/env python3
import rospy
from python.msg import DataType

def callback(data):
    rospy.loginfo(rospy.get_caller_id() + "The number = ", msg.f6_value)
    rospy.loginfo(rospy.get_caller_id() + "The number = ", msg.f5_value)
    rospy.loginfo(rospy.get_caller_id() + "The number = ", msg.i6_value)
    rospy.loginfo(rospy.get_caller_id() + "The number = ", msg.i5_value)
    
def subscriber_msg():
    rospy.init_node('subscriber_msg', anonymous=True)
    rospy.Subscriber("chatter/msg", DataType, callback)
    rospy.spin()

if __name__ == '__main__':
    subscriber_msg()

File: cpp/src/subscriber/subscriber_msg.cpp
#include "ros/ros.h"
#include "cpp/DataType.h"

void chatterRandomCallback(const cpp::DataType::ConstPtr& msg)
{
  ROS_INFO("The float64 value = ", msg->f6_value);
  ROS_INFO("The float32 value = ", msg->f5_value);
  ROS_INFO("The int64 value = ", msg->i6_value);
  ROS_INFO("The int32 value = ", msg->i5_value);
}

int main(int argc, char **argv)
{
 
  ros::init(argc, argv, "subscriber_msg");

  ros::NodeHandle n;

  ros::Subscriber sub = n.subscribe("chatter/datatype", 1000, chatterRandomCallback);

  ros::spin();

  return 0;
}
